### PR TITLE
Remove Wan specific map-operation-provider

### PR DESCRIPTION
**Pull request by ahmetmircik**

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/map/impl/operation/MapOperation.java [hunk 1]
```diff
@@ -56,6 +56,7 @@
      * otherwise in active-active scenarios infinite loop of event forwarding can be seen.
      */
     protected boolean disableWanReplicationEvent;
+    private boolean canPublishWanEvent;
```
Comment (blazember) [inline hazelcast/src/main/java/com/hazelcast/map/impl/operation/MapOperation.java]: The field should be marked as transient to stay consistent with the other not serialized fields and to show the intention. Or we should keep the `canPublishWanEvent()` method call below and remove the field.

Comment (ahmetmircik) [reply hazelcast/src/main/java/com/hazelcast/map/impl/operation/MapOperation.java]: done.

**Comment by ahmetmircik** [comment]
@mmedenjak i deleted both `MapOperationProviderDelegator` and its test file. There will be no `MapOperationProviderDelegator` after this fix.

File: hazelcast/src/main/java/com/hazelcast/map/impl/operation/MapOperation.java [hunk 2]
```diff
@@ -55,7 +55,8 @@
      * Used by wan-replication-service to disable wan-replication event publishing
      * otherwise in active-active scenarios infinite loop of event forwarding can be seen.
      */
-    protected boolean disableWanReplicationEvent;
+    protected transient boolean disableWanReplicationEvent;
```
Comment (blazember) [inline hazelcast/src/main/java/com/hazelcast/map/impl/operation/MapOperation.java]: Opposed to `canPublishWanEvent`, `disableWanReplicationEvent` is serialized in some operations. I think it should be like this:
```
protected boolean disableWanReplicationEvent;
private transient boolean canPublishWanEvent;
```

Comment (ahmetmircik) [reply hazelcast/src/main/java/com/hazelcast/map/impl/operation/MapOperation.java]: Oops, should be ok now.

**Pull request merged**